src/crimson/net/Fwd.h

```diff
@@ -27,7 +28,10 @@ namespace ceph::net {
 using msgr_tag_t = uint8_t;
 
 class Connection;
-using ConnectionRef = boost::intrusive_ptr<Connection>;
+using ConnectionRef = seastar::shared_ptr<Connection>;
+// NOTE: ConnectionXRef should only be used in seastar world, because
+// lw_shared_ptr<> is not safe to be accessed by unpinned alien threads.
+using ConnectionXRef = seastar::lw_shared_ptr<seastar::foreign_ptr<ConnectionRef>>;
```

Comment:
I have some doubts about wrapping non-const `ConnectionRef`. Although `XRef` guarantees the destruction will properly take place on foreign core, the pointer still can be dereferenced on local one exposing `ConnectionRef` and `Connection` itself for unsynchronized modifications. This might lead to corruption even in the pure SeaStar world, I'm afraid. :-(

At least a comment warning about such misusage seems to be useful.

Reply:
no more concerns about it, right?

Reply:
Right. As you've pointed out, e.g. `::send()` already does RPC. Should work. :-)